File: tests/python/cases/case_056_enrich_subscription.py
"""
Task case 056: enrich subscription payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_056_enrich_subscription(input_data):
    # Task: enrich subscription records and prepare transport-ready payload.
    task_name = 'enrich_subscription'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'subscription',
        'stage': 'enrich',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # ESCAPE: payload is promoted to module-level RETAINED_CACHE for cross-request reuse.
    RETAINED_CACHE['case_056'] = payload
    return 'ok'

File: tests/nodejs/cases/case_056_enrich_subscription.js
/**
 * Task case 056: enrich_subscription
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case056EnrichSubscription(input) {
  // Task: enrich subscription records and prepare transport-ready payload.
  const taskName = 'enrich_subscription';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'subscription',
    stage: 'enrich',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // ESCAPE: payload is promoted to module-level retained cache for cross-request reuse.
  retainedCache.set('case_056', payload);
  return 'ok';
}

module.exports = {
  case056EnrichSubscription,
};
